core/controllers/reader.py

```diff
@@ -213,6 +213,25 @@ def get(self, exploration_id):
 
         exploration_rights = rights_manager.get_exploration_rights(
             exploration_id, strict=False)
+
+        # Retrieve all classifiers for the exploration.
+        state_classifier_mapping = {}
+        classifier_training_jobs = (
+            classifier_services.get_classifier_training_jobs(
+                exploration_id, exploration.version, exploration.states))
+        for index, state_name in enumerate(exploration.states):
+            if classifier_training_jobs[index] == None:
+                state_classifier_mapping[state_name] = None
+            else:
+                # Not making this a multi-get since we are deprecating
```

Comment:
Yes, we should avoid GETs in a for loop, so this does need to be rewritten.

However, I was just wondering -- in this PR, can't you do get_multi() on the job results models without deprecating the ClassifierDataModels? I.e., you could just ignore/bypass the latter for now, and deprecate them later? (But if I'm misunderstanding and you do need to get rid of them for this to work, then yep, we'll need to remove them.)